### ISSUE TITLE
Add Universal support

### ISSUE THREAD

**Issue by mrjmd** [contentacms/contenta_angular#2]
https://medium.com/@evertonrobertoauler/angular-4-universal-app-with-angular-cli-db8b53bba07d

**Comment by mrjmd** [comment]
Just for anyone who stumbles across this issue, this is still on our roadmap but is probably not going to land any time soon. Upon further research the method in the Medium article above is not really the best approach, and you end up not having some of the best features of both the CLI and Universal with that approach.

The Angular team is still working on adding Universal support to the CLI, but it sounds like it's a relatively high-effort task that may take a while. In the meantime, our best option may be to have a Universal branch of this project where we `ng eject` from the CLI and then customize our webpack config manually.

**Comment by joaogarin** [comment]
I agree with having a universal branch on the repo. but its something we can do once we feel the app is sort of ready for use. I think it can be accomplished easily so no point really in creating in now and having to do some maybe hard rebases to keep it in sync while we actively change the solution drastically..But definitely want to point out the is a super priority issue for us.

**Comment by mrjmd** [comment]
There is some progress being made on CLI integration, an early (alpha / experimental / branch) version of it may land as soon as next week in time for the talk about Angular on the Server at next week's [Node Summit](http://www.nodesummit.com/agenda).

**Comment by joaogarin** [comment]
thats great news ! :tada:

**Comment by mrjmd** [comment]
Experimental CLI / Universal support has landed: https://twitter.com/hanslatwork/status/888527770925506560

**Comment by mrjmd** [comment]
Big news on this one, some new [documentation](https://github.com/angular/angular-cli/wiki/stories-universal-rendering) and a new [post](https://medium.com/@feloy/creating-an-angular-universal-app-with-the-angular-cli-5ef26c9fd9a5) on Medium have opened the possibilities up here.

We've got a work in progress branch / PR at #73.

**Comment by mrjmd** [comment]
I think our blocker soon is going to be hosting. I've done a little bit of digging into how Universal would work on Firebase, and there is some possibility of using Cloud Functions, but I haven't found any good guides.

**Comment by mrjmd** [comment]
Next update re: cloud functions, here's an example using Universal with Firebase Cloud Functions:
